opencodeblocks/blocks/block.py

```diff
@@ -319,3 +332,13 @@ def deserialize(self, data: dict, hashmap: dict = None, restore_id=True) -> None
                     hashmap.update({socket_data["id"]: socket})
 
         self.update_all()
+
+    def complete_with_default(self, data: OrderedDict) -> None:
+        """Add default data in place when fields are missing"""
+        for key in NONE_OPTIONAL_FIELDS:
+            if key not in data:
+                raise ValueError(f"{key} of the socket is missing")
+
+        for key in DEFAULT_BLOCK_DATA:
+            if key not in data:
+                data[key] = DEFAULT_BLOCK_DATA[key]
```

Comment:
Using global constants lead to a triple duplicate of this same code, a good example of why constants should be use with care !

Reply:
I still feel like this is a duplicate, am I missing something ?